kvmagent/kvmagent/plugins/vm_metadata/rw.py

```
@@ -0,0 +1,678 @@
+def _read_and_parse_slot(fd, offset, capacity, strict=True):
+    """Read slot data with optimistic 1-MB-first strategy (§6.4.1)."""
+    read_size = min(capacity, OPTIMISTIC_READ_SIZE) if capacity > 0 \
+        else OPTIMISTIC_READ_SIZE
+    data = aligned_pread(fd, read_size, offset)
+
+    slot = parse_slot(data, expected_offset=offset,
+                      expected_capacity=capacity if strict else None,
+                      strict=strict)
+
+    if not slot.valid and read_size < capacity:
+        # Payload might be larger than our first read – check the header
```

Ruff (0.15.2)

[warning] Comment contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF003)